extensions/wikia/AuthModal/js/AuthModal.js

```diff
@@ -25,9 +25,15 @@ define('AuthModal', ['jquery', 'wikia.window'], function ($, window) {
 			label: 'username-login-modal'
 		});
 
-		$(window.document).keyup(onKeyUp);
-		window.addEventListener('message', function (event) {
-			if (event.data.isUserAuthorized) {
+		$(window).on('keyup.authModal', document, onKeyUp);
+		$(window).on('message.authModal', function (event) {
+			var e = event.originalEvent;
+
+			if (typeof e.data === 'undefined' || typeof e.data.isUserAuthorized === 'undefined') {
```

Comment:
Is the second check needed? On line 36 you'll verify isUserAuthorized and everything else in this handler is inside the second "if"'s body.